### PR TITLE
chore: fix backend lint errors

### PR DESCRIPTION
**Pull request by filp**
See #9286 for context.

### Description:

Fixes eslint errors for `packages/backend`. Mostly via `lint --fix`, with some manual fixes in the mix.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/clients/EmailClient/EmailClient.ts [hunk 1]
```diff
@@ -165,7 +165,8 @@ export default class EmailClient {
                 roleAction = 'manage';
                 break;
             default:
-                const nope: never = projectMember.role;
+                // NOOP - Intentionally left blank
```
Comment (IrakliJani) [inline packages/backend/src/clients/EmailClient/EmailClient.ts:168]: Hey @filp, This call checks whether the switch/case is "exhaustive" or not. Was this change esaential?

Fyi, We can use the `assertUnreachable` utility function from common.

File: packages/backend/src/clients/EmailClient/EmailClient.test.ts [hunk 2]
```diff
@@ -41,6 +43,8 @@ describe('EmailClient', () => {
             );
         });
         test('should create transported with secure connection when using port 465', async () => {
+            // Relies on side-effects from instancing EmailClient
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
             const client = new EmailClient({
```
Comment (IrakliJani) [inline packages/backend/src/clients/EmailClient/EmailClient.test.ts:48]: maybe we can use `client_` or `_client` don't remember which one

File: packages/backend/src/services/OrganizationService/OrganizationService.ts [hunk 3]
```diff
@@ -110,7 +110,7 @@ export class OrganizationService {
     }
 
     async updateOrg(
-        { organizationUuid, organizationName, userUuid, ability }: SessionUser,
```
Comment (IrakliJani) [inline packages/backend/src/services/OrganizationService/OrganizationService.ts:113]: unused?

File: packages/backend/src/projectAdapters/projectAdapter.ts [hunk 4]
```diff
@@ -118,7 +118,6 @@ export const projectAdapterFromConfig = async (
                 useDbtLs,
             });
         default:
-            const never: never = config;
```
Comment (IrakliJani) [inline packages/backend/src/projectAdapters/projectAdapter.ts:121]: same here

File: packages/backend/src/models/DashboardModel/DashboardModel.ts [hunk 5]
```diff
@@ -186,7 +178,6 @@ export class DashboardModel {
                     });
                     break;
                 default: {
-                    const never: never = tile;
```
Comment (IrakliJani) [inline packages/backend/src/models/DashboardModel/DashboardModel.ts:189]: one more

File: packages/backend/src/logging/winston.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import { SessionUser } from '@lightdash/common';
-import * as express from 'express';
+import { type SessionUser } from '@lightdash/common';
+import type * as express from 'express';
```
Comment (IrakliJani) [inline packages/backend/src/logging/winston.ts:2]: how does this work?

File: packages/backend/src/clients/EmailClient/EmailClient.ts (hunk 1 above)
Comment (filp) [reply packages/backend/src/clients/EmailClient/EmailClient.ts:168]: Was this supposed to raise an `eslint` error? I agree with using `assertUnreachable`, but didn't know it was available under `common` and didn't want to introduce new exceptions as part of this changeset.

Agreed overall though, I'll make the adjustments!

File: packages/backend/src/clients/EmailClient/EmailClient.test.ts (hunk 2 above)
Comment (filp) [reply packages/backend/src/clients/EmailClient/EmailClient.test.ts:48]: Good point, will fix!

File: packages/backend/src/services/OrganizationService/OrganizationService.ts (hunk 3 above)
Comment (filp) [reply packages/backend/src/services/OrganizationService/OrganizationService.ts:113]: Correct, though in retrospect I'll re-introduce this and tag it as unused instead, to keep with my own rule of not modifying implementation too much.

File: packages/backend/src/clients/EmailClient/EmailClient.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/clients/EmailClient/EmailClient.ts:168]: good point! if we don't want to raise then maybe we could do `_nope` or `nope_`?
but then again, looks like a code block that should never ever happen and its better to throw in that case

**Comment by filp** [comment]
Closing until I have more bandwidth to tackle this + include other proposed changes.

**Pull request closed without merging**